tempodb/encoding/vparquet2/nested_set_model.go

```diff
@@ -0,0 +1,105 @@
+package vparquet2
+
+import "github.com/grafana/tempo/pkg/util"
+
+func assignNestedSetModelBounds(trace *Trace) {
+	var (
+		assignmentNeeded bool
+		rootSpans        []*wrappedSpan
+		spanChildren     = map[[8]byte][]*Span{}
+	)
+
+	// find root spans and map span IDs to child spans
+	for _, rs := range trace.ResourceSpans {
+		for _, ss := range rs.ScopeSpans {
+			for i, s := range ss.Spans {
+				if s.NestedSetLeft == 0 || s.NestedSetRight == 0 {
```

Comment:
this seems to be checking for a case where we have combined two traces and some spans already have NestedSetLeft/Right set. can we trust the values that exist tho? this seems to trust existing NestedSetLeft and NestedSetRight values but if we combine two existing traces don't we have to completely recalculate all left/right/parent ids?

Reply:
You are right. Combined traces can have incorrect bounds while all spans have non-zero left and right values. I added a parameter `forceAssignment`. When the function is called in the combiner, the assignment is now enforced.

Reply:
we're forcing reassignment in the combiner and in traceToParquet we're guaranteed that all NestedSet's will be 0. do we even need this check anymore?

Reply:
You are right. I removed the check and the `forceAssignment` parameter